### PR TITLE
Refs #6853 operator[] const and cast methods.

### PR DESCRIPTION
**Pull request by LuisGP**
This PR implements some features needed by `utilities.hpp`

### CODE REVIEW

File: include/xtypes/DynamicData.hpp [hunk 1]
```diff
@@ -864,6 +866,196 @@ class DynamicData : public WritableDynamicDataRef
     /// \brief Request a writable reference from this DynamicData.
     /// \returns a WritableDynamicDataRef identifying th DynamicData.
     WritableDynamicDataRef ref() const { return WritableDynamicDataRef(*this); }
+
+    template<typename T, class = std::is_arithmetic<T>>
```
Comment (lemunozm) [inline include/xtypes/DynamicData.hpp]: Ok, I have several comments about it:
- Could you add an assert for primitive types?
- It is really need to use the switch?, Maybe a: `*static_cast<T*>(this->instance)` could be enough as you implement in last line of this PR.
- A design comment: I think that his feature is out from this scope. This is for me a soss utility, not a DynamicData utility. If an xtypes user want to cast a DynamicData. He can retrieve the value and cast it by himself.

File: include/xtypes/DynamicData.hpp [hunk 2]
```diff
@@ -864,6 +866,196 @@ class DynamicData : public WritableDynamicDataRef
...
+                std::wstring temp = *this;
+                return reinterpret_cast<T>(temp);
+            }
+            */
+        }
+        return T();
+    }
+
+    template<typename T = std::string>
+    T cast() const
```
Comment (lemunozm) [inline include/xtypes/DynamicData.hpp]: I think that can be retrive in one line if you use the instance and perform an static_cast before call to std::to_string()

**Review by lemunozm — APPROVED** [review]

File: include/xtypes/DynamicData.hpp (hunk 1 above)
Comment (LuisGP) [reply include/xtypes/DynamicData.hpp]: Comments addressed in person :).

File: include/xtypes/DynamicData.hpp (hunk 2 above)
Comment (LuisGP) [reply include/xtypes/DynamicData.hpp]: Comments already addressed.

**Review by lemunozm — APPROVED** [review]

**Pull request merged**